### PR TITLE
Display cumulative setup seconds on the config info page

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Show users why their setup/startup is slow.

Screen Shot 2021-03-25 at 8 46 55 AM

Requires https://github.com/home-assistant/core/pull/48274

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/info/integrations-card.ts [hunk 1]
```diff
@@ -95,8 +105,11 @@ class IntegrationsCard extends LitElement {
                                   </a>
                                 </td>
                               `
-                            : ""}
+                            : html`<td></td>`}
                         `}
+                    ${setup && setup?.seconds
```
Comment (bramkragten) [inline src/panels/config/info/integrations-card.ts:110]:
```suggestion
                    ${setup?.seconds
```

File: src/data/integration.ts [hunk 2]
```diff
@@ -38,3 +43,6 @@ export const fetchIntegrationManifest = (
   hass: HomeAssistant,
   integration: string
 ) => hass.callWS<IntegrationManifest>({ type: "manifest/get", integration });
+
+export const fetchIntegrationSetups = (hass: HomeAssistant) =>
+  hass.callWS<IntegrationSetup[]>({ type: "integration/setup" });
```
Comment (bdraco) [inline src/data/integration.ts:48]:
```suggestion
  hass.callWS<IntegrationSetup[]>({ type: "integration/setup_info" });
```

**Review by spacegaier — APPROVED** [review]
LGTM

File: src/panels/config/info/integrations-card.ts [hunk 3]
```diff
@@ -95,8 +105,11 @@ class IntegrationsCard extends LitElement {
                                   </a>
                                 </td>
                               `
-                            : ""}
+                            : html`<td></td>`}
                         `}
+                    ${setup?.seconds
+                      ? html`<td>${setup.seconds.toFixed(2)}s</td>`
```
Comment (spacegaier) [inline src/panels/config/info/integrations-card.ts:111]:
```suggestion
                      ? html`<td>${setup.seconds.toFixed(2)} s</td>`
```

Just saw one thing: We usually have a blank before the unit in the frontend.

**Comment by bramkragten** [comment]
We should probably look at hiding some columns for mobile or folding the rows, as it already doesn't fit now, and we are adding another column.

**Comment by bdraco** [comment]
> image

That looks good. 👍  I'll try to find an example in the frontend code base and replicate it. Marking this as a draft in the mean time since its too late here to make progress.

**Comment by bdraco** [comment]
I haven't found a modern example of hiding this in the codebase.  There are some references to a `narrow` property, but I'm not sure what the preferred approach is here.

**Comment by bramkragten** [comment]
Yes, we should use `narrow`, for the layout we probably wanna use flexbox.

I can push something if you want tomorrow?

**Comment by bdraco** [comment]
> Yes, we should use `narrow`, for the layout we probably wanna use flexbox.
> 
> I can push something if you want tomorrow?

That would be fantastic!

**Review by bramkragten — APPROVED** [review]

**Comment by bdraco** [comment]
Thanks for the help getting this one  done. 👏👍🎉

**Pull request merged**